### PR TITLE
Xr experience helper

### PR DESCRIPTION
**Pull request by TrevorDev**
Testing PG: http://playground.babylonjs.com#GVAEUW#35

### CODE REVIEW

File: src/Cameras/XR/babylon.webXRExperienceHelper.ts [hunk 1]
```diff
@@ -0,0 +1,116 @@
+module BABYLON {
+    /**
+     * Helper class used to enable XR
+     * @see https://doc.babylonjs.com/how_to/webxr
+     */
+    export class WebXRExperienceHelper {
+        /**
+         * Container which stores the xr camera and controllers as children. This can be used to move the camera/user as the camera's position is updated by the xr device
+         */
+        public container: AbstractMesh;
```
Comment (deltakosh) [inline src/Cameras/XR/babylon.webXRExperienceHelper.ts]: smart!

File: src/Cameras/XR/babylon.webXRExperienceHelper.ts [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+                    // Restore scene settings
+                    this.scene.autoClear = this._originalSceneAutoClear;
+                    this.scene.activeCamera = this._nonVRCamera;
+                    this._sessionManager.onXRFrameObservable.clear();
+                    this._removeCanvas();
+
+                    this.isInStateTransition = false;
+                    this.isInXRMode = false;
+                });
+                this.isInStateTransition = false;
```
Comment (deltakosh) [inline src/Cameras/XR/babylon.webXRExperienceHelper.ts]: I think we should add observables to track the xr lifetime

Comment (TrevorDev) [reply src/Cameras/XR/babylon.webXRExperienceHelper.ts]: good, idea I also removed isInStateTransition  and isInXRMode in favor of a single state enum

File: src/Cameras/XR/babylon.webXRSessionManager.ts [hunk 3]
```diff
@@ -4,17 +4,26 @@ module BABYLON {
      * @see https://doc.babylonjs.com/how_to/webxr
      */
     export class WebXRSessionManager {
-        private _xrNavigator: any;
-        private _xrDevice: XRDevice;
-        private _tmpMatrix = new BABYLON.Matrix();
+        /**
+         * Fires every time a new xrFrame arrives which can be used to update the camera
+         */
+        public onXRFrameObservable: Observable<any> = new BABYLON.Observable<any>();
```
Comment (deltakosh) [inline src/Cameras/XR/babylon.webXRSessionManager.ts]: you need to dispose them

File: src/Cameras/XR/babylon.webXRExperienceHelper.ts [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+
+        /**
+         * The current state of the XR experience (eg. transitioning, in XR or not in XR)
+         */
+        public state: WebXRState = WebXRState.NOT_IN_XR;
+
+        /**
+         * Fires when the state of the experience helper has changed
+         */
+        public onStateChangedObservable = new Observable<WebXRState>();
```
Comment (deltakosh) [inline src/Cameras/XR/babylon.webXRExperienceHelper.ts]: needs to be disposed

File: src/Cameras/XR/babylon.webXRSessionManager.ts [hunk 5]
```diff
@@ -3,18 +3,27 @@ module BABYLON {
...
-        private _tmpMatrix = new BABYLON.Matrix();
+    export class WebXRSessionManager implements IDisposable {
+        /**
+         * Fires every time a new xrFrame arrives which can be used to update the camera
+         */
+        public onXRFrameObservable: Observable<any> = new BABYLON.Observable<any>();
+        /**
+         * Fires when the xr session is ended either by the device or manually done
+         */
+        public onXRSessionEnded: Observable<any> = new BABYLON.Observable<any>();
```
Comment (deltakosh) [inline src/Cameras/XR/babylon.webXRSessionManager.ts]: needs to be disposed as well :) (all observables)

**Pull request merged**